src/main/asciidoc/batch/file-ingest-sftp/kubernetes.adoc

```diff
@@ -0,0 +1,278 @@
+
+The source code for the <<Batch File Ingest>> batch job is located in `batch/file-ingest`.
+We will need to build a Docker image for this app and publish it to a Docker registry
+accessible to your Kubernetes cluster. 
+For your convenience, the Docker image is available at 
+https://hub.docker.com/r/springcloud/ingest/[springcloud/ingest].
+
+. Build and publish the Docker image
++
+Skip this step if you are using the pre-built image. We are using the `fabric8` Maven docker plugin. which will push images to https://hub.docker.com/[Docker Hub] by default. You will need to have a Docker Hub account for this. Note the `-Pkubernetes` flag adds a dependency to provide the required Maria DB JDBC driver. 
++
+[source,console,options=nowrap]
+----
+$cd batch/file-ingest
+$mvn clean package docker:build docker:push -Ddocker.org=<DOCKER_ORG> -Ddocker.username=<DOCKER_USERNAME> -Ddocker.password=<DOCKER_PASSWORD> -Pkubernetes
+----
++  
+. Create the remote directory
++
+Create a directory on the SFTP server where the `sftp` source will detect files and download them for processing.
+This path must exist prior to running the demo and can be any location that is accessible by the configured SFTP user.
+On the SFTP server create a directory called `remote-files`, for example:
++
+```
+sftp> mkdir remote-files
+```
++
+. Create a shared NFS directory
++
+Create a read/write directory on the NFS server.
++
+```
+$ sudo mkdir /export/shared-files
+$ sudo chmod 0777 /export/shared-files
+```
+. Register the `sftp-dataflow` source and the `tasklauncher-dataflow` sink
++
+With our Spring Cloud Data Flow server running, we register the `sftp-dataflow` source and `task-launcher-dataflow` sink.
+The `sftp-dataflow` source application will do the work of polling the remote directory for new files and downloading them to the local directory.
+As each file is received, it emits a message for the `task-launcher-dataflow` sink to launch the task to process the data from that file.
++
+In the Spring Cloud Data Flow shell:
++
+[source,console,options=nowrap]
+----
+dataflow:>dataflow:>app register --name sftp --type source --uri docker:springcloud/sftp-dataflow-source-kafka --metadata-uri maven://org.springframework.cloud.stream.app:sftp-dataflow-source-kafka:jar:metadata:2.1.0.BUILD-SNAPSHOT
+Successfully registered application 'source:sftp'
+dataflow:>app register --name task-launcher --type sink --uri docker:springcloud/task-launcher-dataflow-sink-kafka --metadata-uri maven://org.springframework.cloud.stream.app:task-launcher-dataflow-sink-kafka:jar:metadata:1.0.0.BUILD-SNAPSHOT
+Successfully registered application 'sink:task-launcher'
+----
++
+. Register and create the file ingest task:
+[source,console,options=nowrap]
+dataflow:>app register --name fileIngest --type task --uri docker:springcloud/ingest
+Successfully registered application 'task:fileIngest'
+dataflow:>task create fileIngestTask --definition fileIngest
+Created new task 'fileIngestTask'
```

Comment:
Why do we need to create a task? Won't the `fileIngest` app directly used in the stream definition instead?